plugin/src/App/StatusPages/NotConnected.lua

```diff
@@ -0,0 +1,156 @@
+local Rojo = script:FindFirstAncestor("Rojo")
+local Plugin = Rojo.Plugin
+
+local Roact = require(Rojo.Roact)
+
+local Config = require(Plugin.Config)
+
+local Theme = require(Plugin.App.Theme)
+local BorderedContainer = require(Plugin.App.Components.BorderedContainer)
+local TextButton = require(Plugin.App.Components.TextButton)
+local Header = require(Plugin.App.Components.Header)
+
+local PORT_WIDTH = 74
+local DIVIDER_WIDTH = 1
+local HOST_OFFSET = 12
+
+local e = Roact.createElement
+
+local function AddressEntry(props)
+	return Theme.with(function(theme)
+		return e(BorderedContainer, {
+			transparency = props.transparency,
+			size = UDim2.new(1, 0, 0, 36),
+			layoutOrder = props.layoutOrder,
+		}, {
+			Host = e("TextBox", {
+				Text = "",
+				Font = Enum.Font.Code,
+				TextSize = 18,
+				TextColor3 = theme.AddressEntry.TextColor,
+				TextXAlignment = Enum.TextXAlignment.Left,
+				TextTransparency = props.transparency,
+				PlaceholderText = Config.defaultHost,
+				PlaceholderColor3 = theme.AddressEntry.PlaceholderColor,
+
+				Size = UDim2.new(1, -(HOST_OFFSET + DIVIDER_WIDTH + PORT_WIDTH), 1, 0),
+				Position = UDim2.new(0, HOST_OFFSET, 0, 0),
+
+				ClipsDescendants = true,
+				BackgroundTransparency = 1,
+
+				[Roact.Ref] = props.hostRef,
+			}),
+
+			Port = e("TextBox", {
+				Text = "",
+				Font = Enum.Font.Code,
+				TextSize = 18,
+				TextColor3 = theme.AddressEntry.TextColor,
+				TextTransparency = props.transparency,
+				PlaceholderText = Config.defaultPort,
+				PlaceholderColor3 = theme.AddressEntry.PlaceholderColor,
+
+				Size = UDim2.new(0, PORT_WIDTH, 1, 0),
+				Position = UDim2.new(1, 0, 0, 0),
+				AnchorPoint = Vector2.new(1, 0),
+
+				BackgroundTransparency = 1,
+
+				[Roact.Ref] = props.portRef,
+
+				[Roact.Change.Text] = function(object)
+					local text = object.Text
+
+					text = text:gsub("%D", "")
+					text = text:sub(1, 5)
```

Comment:
Even though ports don't go over 2¹⁶, truncating text like this is usually frustrating UX, and we should avoid doing it!

Reply:
The UI isn't really designed to support more than 5 characters. Are you fine with it just overflowing?